publication-rest/src/test/java/no/unit/nva/publication/update/PublishPublicationHandlerTest.java

```diff
@@ -0,0 +1,175 @@
+import static no.unit.nva.publication.PublicationServiceConfig.dtoObjectMapper;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.Publication;
+import no.unit.nva.publication.model.business.Resource;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.ResourcesLocalTest;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.stubs.FakeContext;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+class PublishPublicationHandlerTest extends ResourcesLocalTest {
+
+    private Context context;
+    private ByteArrayOutputStream output;
+    private ResourceService resourceService;
+    private PublishPublicationHandler handler;
+
+    @BeforeEach
+    void setUp() {
+        super.init();
+        context = new FakeContext();
+        output = new ByteArrayOutputStream();
+        resourceService = getResourceServiceBuilder().build();
+        handler = new PublishPublicationHandler(resourceService);
+    }
+
+    @Test
+    void shouldReturnUnauthorizedWhenUserIsNotAuthorized() throws IOException {
+        var publicationIdentifier = SortableIdentifier.next();
+        var request = createUnauthorizedRequest(publicationIdentifier);
+        handler.handleRequest(request, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertEquals(HTTP_UNAUTHORIZED, response.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenPublicationDoesNotExist() throws IOException {
+        var publicationIdentifier = SortableIdentifier.next();
+        var request = createRequestWithRandomUser(publicationIdentifier);
+        handler.handleRequest(request, output, context);
+
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertEquals(HTTP_NOT_FOUND, response.getStatusCode());
+    }
+
+    @Test
+    void shouldReturnForbiddenWhenUserHasNoPermissionsToPublishPublication() throws IOException, BadRequestException {
+        var publication = createPublication();
+        var request = createRequestWithRandomUser(publication.getIdentifier());
```

Comment:
random user implies that it will occasionally have the correct permissions to publish…

Reply:
renaming to `createRequestWithUserWithoutPermissions()`